src/adapter/src/catalog.rs

```diff
@@ -2714,7 +2714,17 @@ impl Catalog {
         // table updates and most-recent-timestamp calculations on a single
         // iterator.
         let storage_usage_events = catalog.storage().await.storage_usage().await?;
-        for event in storage_usage_events {
+        for event in
+            storage_usage_events.filter(|event| match config.storage_usage_retention_period {
+                None => true,
+                Some(period) => {
+                    let period_millis = period.as_millis();
+                    let moment = (config.now)();
+                    let VersionedStorageUsage::V1(e) = event;
```

Comment:
`event.timestamp()` is a forward-compatible way to do this this is guaranteed to exist for any future version.